meshpy/four_c/run_four_c.py

```diff
@@ -0,0 +1,147 @@
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+# -----------------------------------------------------------------------------
+"""Provide a function that allows to run 4C"""
+
+
+import os
+import sys
+import subprocess
+
+
+def run_four_c(
+    input_file,
+    output_dir,
+    *,
+    four_c_exe=None,
+    mpi_command=None,
+    n_proc=None,
+    output_name="xxx",
+    restart_step=None,
+    restart_from=None,
+    log_to_console=False,
+):
+    """Run a 4C simulation and return the exit code of the run
+
+    This function looks into the environment variables for some parameters:
+        "MESHPY_FOUR_C_EXE"
+        "MESHPY_MPI_COMMAND"
+        "MESHPY_MPI_NUM_PROC"
+    If the corresponding keyword arguments are set, they overwrite the environment
+    variable.
+
+    Args
+    ----
+    input_file: str
+        Path to the dat file on the filesystem
+    output_dir: str
+        Directory where the simulation should be performed (will be created if
+        it does not exist)
+    four_c_exe: str
+        Optionally explicitly specify path to the 4C executable
+    mpi_command: str
+        Command to launch MPI, defaults to "mpirun"
+    n_proc: int
+        Number of process used with MPI, defaults to 1
+    output_name: str
+        Base name of the output files
+    restart_step: int
+        Time step to restart from
+    restart_from: str
+        Path to initial simulation (relative to output_dir)
+    log_to_console: bool
+        If the 4C simulation output should be shown in the console.
+
+    Return
+    ----
+    return_code: int
+        Return code of 4C run
+    """
+
+    def get_env_variable(name, *, fail_if_not_set=False, default=None):
+        """Return the value of an environment variable"""
+        if name in os.environ.keys():
+            return os.environ[name]
+        elif fail_if_not_set:
+            raise ValueError(f"Environment variable {name} is not set")
+        return default
```

Comment:
Should we consider moving this function to a more central place?

Reply:
Good idea, I will do that.